MaxText/layers/linears.py

```diff
@@ -253,38 +290,114 @@ def dense_general(
       parameter_memory_host_offload=parameter_memory_host_offload,
       name=name,
       metadata_fn=variable_to_logically_partitioned,
+      abstract_init=False,
   )
   return module
 
 
-class MlpBlock(nn.Module):
-  """Transformer MLP / feed-forward block.
-
-  Attributes:
-    intermediate_dim: Shared dimension of hidden layers.
-    activations: Type of activations for each layer.  Each element is either
-      'linear', a string function name in flax.linen, or a function.
-    kernel_init: Kernel function, passed to the dense layers.
-    deterministic: Whether the dropout layers should be deterministic.
-    intermediate_dropout_rate: Dropout rate used after the intermediate layers.
-    dtype: computation data type for the dense layer.
-    weight_dtype: weight data type for the dense layer.
-    use_bias: whether to add bias in all feedforward layers.
-    use_pre_norm: whether to add pre layer norm in mlp layers.
-    quant: Optional quantization config, no quantization if None.
-  """
+class MlpBlock(nnx.Module):
+  """Transformer MLP / feed-forward block."""
+
+  def __init__(
+      self,
+      config: Config,
+      in_features: int,
```

Comment:
Nit: can this be num_features to be consistent with changes we made in other places?